neural_chat/models/base_model.py

```diff
@@ -210,43 +214,29 @@ def get_default_conv_template(self, model_path: str) -> Conversation:
         """
         return get_conv_template("one_shot")
 
-    def register_tts(self, instance):
-        """
-        Register a text-to-speech (TTS) instance.
-
-        Args:
-            instance: An instance of a TTS module.
-        """
-        self.tts = instance
-
-    def register_asr(self, instance):
-        """
-        Register an automatic speech recognition (ASR) instance.
-
-        Args:
-            instance: An instance of an ASR module.
-        """
-        self.asr = instance
-
-    def register_safety_checker(self, instance):
-        """
-        Register a safety checker instance.
-
-        Args:
-            instance: An instance of a safety checker module.
-        """
-        self.safety_checker = instance
-
-    def register_retriever(self, retriever, retrieval_type):
+    def register_plugin(self, plugin_name, instance):

```

Comment:
suggest to change this func name to another to avoid mix up with same name decorator.